### ISSUE TITLE
Improve accessibility/usability of the transcript interface

### ISSUE THREAD

**Issue by terrill** [ableplayer/ableplayer#538]
Feedback from @kb-smarterbalanced, originally in #499: 

> Issue 6:
> Transcript Modal > Green highlighted link sections of the transcript
> Additional content is only revealed on hover, which is inaccessible to screen reader and users.
> Mouse users are able to hover over the transcript text which then highlights sections of the text. These sections are clickable for mouse users, taking them to that section of the video. Additionally, there is an option to make these sections reachable to keyboard users by selecting the "Keyboard-enable transcript" checkbox in the "Transcript Preferences". However, in neither of these situations are these actionable to screen reader users.
> 
> Issue 7:
> Transcript Modal > Yellow highlighted link section of the transcript (indicates current section of the transcript)
> This element uses specific sensory characteristics, such as color alone, to indicate that it is selected. This information will not be available to all users.
> Ensure that this element uses other means to convey its status to sighted users such as an underline. For screen reader users, this element will also need its status conveyed by the addition of aria-current="true" attribute when selected.
> 
> 

In addition to those issues, we need to do more user testing of the transcript with screen reader users and keyboard users. It's currently challenging to jump back and forth between the player and the transcript.

**Comment by terrill** [comment]
Transcript highlight styles have been updated in c1916faf6f5a378a3a87c45c454e849e9a83307b to eliminate color dependency. 

As for screen reader accessibility, I welcome others' input as to how best to communicate the interactive components of the transcript. Adding ARIA to explicitly identify clickable components within the transcript, as well as which component is currently highlighted, would seriously impede usability for screen reader users. The transcript should be first and foremost readable as a transcript.  I've surveyed a couple of screen reader users for their opinions, and both suggested making no changes. However, I'm still open to considering creative solutions.  

One other thought to consider as we contemplate possible solutions:  There's actually no obvious indication for *any* users that the transcript is clickable. All users, with or without eyesight, are expected to discover that.  Sighted users do get a visual hint of its interactivity if they hover over transcript text, so there is that disparity.  Anyway, I welcome ideas...

**Comment by kb-smarterbalanced** [comment]
Thank you so much for your work on this! I will run this question by our UX team and respond.

**Comment by terrill** [comment]
I did some brainstorming this week with a colleague who uses a screen reader. We came up with an idea for improving the accessibility of the transcript without compromising its usability. We could offer three modes, accessible via a select field in the toolbar at the top of the Transcript  window, with keyboard shortcuts available for switching between them on-the-fly: 

- "Reading mode" - The default mode, no semantics that would result in extra verbiage for screen reader users, so they can simply read the transcript as text.  Also, transcript components are not part of the tab order so keyboard users can easily tab past the transcript. 
- "Keyboard mode" - Add `tabindex="0"` to all transcript components so keyboard users can tab through the transcript and click an item by pressing Enter (this is currently already supported via the Transcript Preferences menu) 
- "Interactive mode" - Add `role="button"` to all clickable items, and `aria-current="true"` to the component that's currently highlighted, so if screen reader users want the extra semantics it's available to them.  

Another idea is to include brief visible instructions at the top of the transcript so all users know it's interactive, or can be made so by changing the mode. 

All this will require new text to be added, which will require translation (we're now up to 19 languages!) so it's a longer term project and won't go into the next release.

**Comment by kb-smarterbalanced** [comment]
Hi Terrill, this is amazing, we're working here with screen reader users as well. This sounds like a great approach and we are looking forward to seeing how you develop it. Let us know when it's released, I hear you that it's longer term, we would be happy to help test then. Ok to close this issue as you're adding it as a new feature.
